### ISSUE TITLE
Failed to write MSR

### ISSUE THREAD

**Issue by dergoegge** [awslabs/snapchange#3]
I have managed to get snapchange to run on the `01_getpid` example, however only after commenting out this line:

https://github.com/awslabs/snapchange/blob/a442cd39b13a71e721e60f242837324149cd2206/src/fuzzvm.rs#L3568

With the assertion in place the output looks like this:

<details>
<summary>$ RUST_BACKTRACE=1 cargo +nightly run -- fuzz</summary>

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
     Running `target/debug/fuzzer_template fuzz`
Using project config: Config {
    stats: Stats {
        merge_coverage_timer: 60s,
        maximum_new_corpus_size: 250,
        minimum_total_corpus_percentage_sync: 10,
        maximum_total_corpus_percentage_sync: 50,
    },
    guest_memory_size: 5368709120,
}
Thread 0 returned err.. Failed to write MSR

Stack backtrace:
   0: anyhow::error::<impl anyhow::Error>::msg
             at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.71/src/error.rs:83:36
   1: anyhow::__private::format_err
             at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.71/src/lib.rs:669:13
   2: snapchange::fuzzvm::FuzzVm<FUZZER>::restore_guest_msrs
             at /root/snapchange/src/fuzzvm.rs:3568:9
   3: snapchange::fuzzvm::FuzzVm<FUZZER>::init_guest
             at /root/snapchange/src/fuzzvm.rs:3257:21
   4: snapchange::fuzzvm::FuzzVm<FUZZER>::create
             at /root/snapchange/src/fuzzvm.rs:777:9
   5: snapchange::commands::fuzz::start_core
             at /root/snapchange/src/commands/fuzz.rs:669:22
   6: snapchange::commands::fuzz::run::{{closure}}::{{closure}}
             at /root/snapchange/src/commands/fuzz.rs:376:17
   7: std::panicking::try::do_call
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/panicking.rs:485:40
   8: __rust_try
   9: std::panicking::try
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/panicking.rs:449:19
  10: std::panic::catch_unwind
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/panic.rs:142:14
  11: snapchange::commands::fuzz::run::{{closure}}
             at /root/snapchange/src/commands/fuzz.rs:375:26
  12: std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/sys_common/backtrace.rs:135:18
  13: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/thread/mod.rs:529:17
  14: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/core/src/panic/unwind_safe.rs:271:9
  15: std::panicking::try::do_call
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/panicking.rs:485:40
  16: __rust_try
  17: std::panicking::try
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/panicking.rs:449:19
  18: std::panic::catch_unwind
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/panic.rs:142:14
  19: std::thread::Builder::spawn_unchecked_::{{closure}}
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/thread/mod.rs:528:30
  20: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/core/src/ops/function.rs:250:5
  21: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/alloc/src/boxed.rs:1985:9
  22: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/alloc/src/boxed.rs:1985:9
  23: std::sys::unix::thread::Thread::new::thread_start
             at /rustc/5ea3f0ae08c07472239a94ce55601e9b63eb1f45/library/std/src/sys/unix/thread.rs:108:17
  24: start_thread
             at /build/glibc-6iIyft/glibc-2.28/nptl/pthread_create.c:486:8
  25: clone
             at /build/glibc-6iIyft/glibc-2.28/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95
Stats waiting for threads to terminate
All cores are dead..
Stats breaking from finish..
Writing graph data to disk..
Writing corpus (24) to disk: 0.00 MB
[stats] FINISHED!
Stats thread: Ok(())
Kick thread: Ok(())
```
</details>

Some light debugging revealed that only the last two MSRs are not being written:
https://github.com/awslabs/snapchange/blob/a442cd39b13a71e721e60f242837324149cd2206/src/fuzzvm.rs#L3517-L3526

---

System info: 
```
Operating System: Debian GNU/Linux 10 (buster)
Kernel: Linux 6.1.0-9-amd64
Architecture: x86-64
```

**Comment by corydu** [comment]
Great catch! Thanks for the issue.

Are you running on an AMD processor by chance? I need to test if those MSRs are specific to Intel or if they are needed anymore. Does commenting out those MSRs successfully execute the guest VM?

**Comment by dergoegge** [comment]
> Are you running on an AMD processor by chance?

Yes I am, it's a `AMD Ryzen 5 3600`.

> Does commenting out those MSRs successfully execute the guest VM?

Yes that works.

**Comment by corydu** [comment]
Been doing a bit of testing on a few machines and this do look to be superfluous. Thanks for the issue!

Fixed in #5
